### PR TITLE
feat/handle_hex_audiodata

### PR DESCRIPTION
**Pull request by JarbasAl**
allow audio binary data to be sent in hex string format

this enables support for ovos-docker and hivemind, companion PR in ovos-workshop also needed for skills

### CODE REVIEW

File: ovos_audio/service.py [hunk 1]
```diff
@@ -415,31 +420,52 @@ def _resolve_sound_uri(uri: str):
             raise FileNotFoundError(f"{audio_file} does not exist")
         return audio_file
 
+    @staticmethod
+    def _path_from_hexdata(hex_audio, audio_ext=None):
```
Comment (NeonDaniel) [inline ovos_audio/service.py:424]: this should include a docstring including the expected contents/encoding of `hex_audio`

File: ovos_audio/service.py [hunk 2]
```diff
@@ -415,31 +420,52 @@ def _resolve_sound_uri(uri: str):
...
-        audio_ext = message.data.get("audio_ext")  # unused ?
         audio_file = message.data.get("uri") or \
                      message.data.get("filename")  # backwards compat
+        hex_audio = message.data.get("binary_data")
+        audio_ext = message.data.get("audio_ext")
+        if hex_audio:
+            audio_file = self._path_from_hexdata(hex_audio, audio_ext)
+
         if not audio_file:
             raise ValueError(f"'uri' missing from message.data: {message.data}")
```
Comment (NeonDaniel) [inline ovos_audio/service.py:451]: This should be updated to something like "no `uri` or `binary_data` in message.data"

File: ovos_audio/service.py [hunk 3]
```diff
@@ -415,31 +420,52 @@ def _resolve_sound_uri(uri: str):
...
         if not validate_message_context(message):
             LOG.debug("ignoring playback, message is not from a native source")
             return
         audio_file = message.data.get("uri")
+        hex_audio = message.data.get("binary_data")
+        audio_ext = message.data.get("audio_ext")
+        if hex_audio:
+            audio_file = self._path_from_hexdata(hex_audio, audio_ext)
         if not audio_file:
             raise ValueError(f"'uri' missing from message.data: {message.data}")
```
Comment (NeonDaniel) [inline ovos_audio/service.py:470]: This should be updated to something like "no `uri` or `binary_data` in message.data"

File: ovos_audio/service.py [hunk 4]
```diff
@@ -415,31 +420,52 @@ def _resolve_sound_uri(uri: str):
...
+
         if not audio_file:
             raise ValueError(f"'uri' missing from message.data: {message.data}")
         audio_file = self._resolve_sound_uri(audio_file)
         audio_ext = audio_ext or audio_file.split(".")[-1]
         listen = message.data.get("listen", False)
 
         sess_id = SessionManager.get(message).session_id
-        TTS.queue.put((audio_ext, str(audio_file), viseme, sess_id, listen, message))
+        TTS.queue.put((str(audio_file), viseme, sess_id, listen, message))
```
Comment (NeonDaniel) [inline ovos_audio/service.py:457]: This looks like a breaking change https://github.com/OpenVoiceOS/ovos-audio/blob/e20159318a7883465a80a220dba2dd565e6d1317/ovos_audio/playback.py#L208-L219

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

File: ovos_audio/service.py (hunk 4 above)
Comment (JarbasAl) [reply ovos_audio/service.py:457]: no, this is exactly avoiding the warning logs, audio_ext passing is deprecated

```python 
if len(speech_data) == 5 and isinstance(speech_data[-1], Message):  # this is how it is supposed to be done
	data, visemes, listen, tts_id, message = speech_data
else:  # avoiding this
	LOG.warning("it seems you are interfacing with TTS.queue directly, this is not recommended!\n"
	"new expected TTS.queue contents -> data, visemes, listen, tts_id, message")
```

**Pull request merged**